analyzers/src/SonarAnalyzer.Common/Helpers/SonarLintXmlReader.cs

```diff
@@ -0,0 +1,106 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using System.Xml.Serialization;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarAnalyzer.Helpers;
+
+public class SonarLintXmlReader
+{
+    private readonly SonarLintXml sonarLintXml;
+    private readonly Lazy<SonarLintXmlSettingsReader> settings;
+
+    public static readonly SonarLintXmlReader Empty = new(null, LanguageNames.CSharp);
+
+    public SonarLintXmlSettingsReader Settings => settings.Value;
+
+    public SonarLintXmlReader(SourceText originalXml, string language)
+    {
+        sonarLintXml = originalXml == null ? SonarLintXml.Empty : ParseContent(originalXml);
+        settings = new Lazy<SonarLintXmlSettingsReader>(() => new(sonarLintXml, language));
+    }
+
+    private static SonarLintXml ParseContent(SourceText sonarProjectConfig)
+    {
+        try
+        {
+            var serializer = new XmlSerializer(typeof(SonarLintXml));
+            using var sr = new StringReader(sonarProjectConfig.ToString());
+            return (SonarLintXml)serializer.Deserialize(sr);
+        }
+        catch
+        {
+            return SonarLintXml.Empty;
+        }
+    }
+}
+
+public class SonarLintXmlSettingsReader
+{
+    private readonly Lazy<bool> ignoreHeaderComments;
```

Comment:
Instead of Lazy, you can also use `??=`:
[sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUPgMwAERNAwjQN4U3s0AOyUAbhAgCmdYgAYaAfQDOAbjYdqI8QGUaAXgB8kqTQD8utTQBKgiBmUAKAJRzy89tz4Dh+McdPnr9lt44iAnBYARCZmykE2vgoA7DRB4bZ+AL4U3kSEPuR+fvzINADG6jQwggDujNaJ2ewSRfkAdMpV1bWGDU1RNXWNMtXs3inkSUA===)

This saves you a lot of lines here.
It would be even nicer with the [Semi-Auto-Properties; field keyword](https://github.com/dotnet/csharplang/issues/140) which comes in C#12